### DISCUSSION TITLE
Trying to create a liquidity pool on local network

### DISCUSSION THREAD

**Discussion by GuiFlam** [Cyfrin/security-and-auditing-full-course-s23#124] [Q&A]
Hi everyone. I am trying to create a liquidity pool on my local network. Here is what I have done so far:

1. I deploy a WETH token
2. I deploy the uniswap factory
3. I deploy the uniswap router using the factory address and the WETH address
4. I deploy the 2 ERC20 tokens I want to create a liquidity pool for.
5. I mint those 2 tokens to my wallet.
6. I approve the uniswap router to spend my 2 tokens.
7. Before doing this step I verify that the router has enough allowance and I have enough of the 2 tokens in my wallet
8. I call the addLiquidity function

I get this error: The transaction has been reverted to the initial state. Note: The called function should be payable if you send value and the value you send should be less than your current balance.

Am I doing something wrong? What steps am I missing?

Uniswap v2 router: https://docs.uniswap.org/contracts/v2/reference/smart-contracts/router-02

Uniswap Factory v2: https://docs.uniswap.org/contracts/v2/reference/smart-contracts/factory

For WETH and the 2 ERC20 I am using: https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol

This is how I am calling the function: https://imgur.com/a/NzF8EsV

**Comment by PatrickAlphaC** [comment]
Hmm... This question is a bit tricky to answer in it's current state. 

Could you please remember to follow this section to format your questions? https://youtu.be/umepbfKp5rI?t=22464

Also, it seems like this question is unrelated to course material, is that correct?

**Reply by Bystrianyk** [reply]
cool

**Comment by Crypton-56** [comment]
Very interesting

**Comment by cryptofandorin** [comment]
Very interesting
